Release

crisvdn opened the pull request:
This PR adds unit tests for the qik 2s12v10 motorcontroller.
This PR also has a patch for a problem that could exist when putting negative values in the brake function.

ramonvbemmel committed:
Added first test

editted makefile and added test to set motorspeed

crisvdn committed:
Merge branch 'feature-rhino-turn-with-encoder' into feature-unit_tests

crisvdn committed:
Added test case

Added test case to see if the 2 bytes that are sent, are corresponding with the register and the speed.

crisvdn committed:
Added tests for motor_m0 and motor_m1

Multiple tests were added, settings the motor speed as a positive and as a negative value.

crisvdn committed:
Small type error

In the description of test cases, motor_m0 was mentioned while motor_m1 was meant.

crisvdn committed:
Added testing of brake function

Added the testing of the general brake function. Added documentation around it. Also added tests to check if the USART send buffer is empty before and after the certain tests

crisvdn committed:
Added test cases for braking with the motor_controller

Added test cases for the motor controller. Specific tests are: general brake function positive and negative value. Seperate motor brake function positive and negative value.

crisvdn committed:
Small doxy type error

crisvdn committed:
Changed parameter of the brake functions. to uint8_t

The registers of the qik board called by the Brake functions doesn't accept negative values. uint8_t would prevent any issues with potentionally calling the function with a negative value.

crisvdn committed:
Merge branch 'release' into patch-brake_parameter

crisvdn committed:
Merge branch 'release' into feature-unit_tests

crisvdn committed:
Merge branch 'master' into patch-brake_parameter

crisvdn committed:
merge commit

crisvdn committed:
Merge branch 'master' into feature-unit_tests

crisvdn committed:
Makefile error

crisvdn committed:
Merge pull request #31 from R2D2-2019/feature-unit_tests

Feature unit tests

crisvdn committed:
Merge branch 'release' into patch-brake_parameter

crisvdn committed:
Changed brake functions to limit from 0 to 127, edited test cases to match functions

Edited the brake functions to limit the value from 0 to 127. If a negative value is passed, it will convert to a uint8_t and thus positive number. It will be limited between 0 and 127. Thats why the tests had to be adjusted to check with 127.

crisvdn committed:
Added doxygen for brake functions

crisvdn committed:
Merge pull request #32 from R2D2-2019/patch-brake_parameter

Patch brake parameter

florianhumblot commented on code/src/qik_2s12v10.cpp:
```diff
@@ -114,20 +114,29 @@ namespace r2d2::moving_platform {
     }
 
-    void qik_2s12v10_c::brake(const int8_t brake_amount) {
+    void qik_2s12v10_c::brake(uint8_t brake_amount) {
```
Why doesn't this function call the `brake_m0` and `brake_m1` functions?

crisvdn replied:
I agree that this can be done. However, these functions were already implemented by someone else (as this code was already on master) and this was not the scope of this PR.

crisvdn replied:
Changed the function in the last commits. Can you take another look?

PatrickDekker98 reviewed (approved):
nothing to say, clean code and clear documentary

Commit:
Change requests for PR 33

This changes the implementation of the universal brake function to call the brake functions of both motors.

crisvdn committed:
Testcases test according to newly written USART buffer.

Due to the implementation fix of the brake function, test cases will check for the set_speed registers and values, since these functions are called

crisvdn committed:
Merge pull request #34 from R2D2-2019/patch-PR_33

Change requests for PR 33
